src/torch/SIRF_torch.py

```diff
@@ -8,29 +8,31 @@
 # https://github.com/educating-dip/pet_deep_image_prior/blob/main/src/deep_image_prior/torch_wrapper.py
 
 
-def sirf_to_torch(sirf_src, torch_dest, requires_grad=False):
+def sirf_to_torch(sirf_src, device, requires_grad=False):
     if requires_grad:
-        return torch.tensor(sirf_src.as_array(), requires_grad=True).to(torch_dest.device)
+        # use torch.tensor to infer data type
+        return torch.tensor(sirf_src.as_array(), requires_grad=True).to(device)
     else:
-        return torch.tensor(sirf_src.as_array()).to(torch_dest.device)
+        return torch.tensor(sirf_src.as_array()).to(device)
 
 def torch_to_sirf(torch_src, sirf_dest): 
     return sirf_dest.fill(torch_src.detach().cpu().numpy())
 
 
-class _ObjectiveFunctionModule(torch.autograd.Function):
+class _ObjectiveFunction(torch.autograd.Function):
     @staticmethod
     def forward(ctx,
```

Comment:
In order to have a clean implementation, it might be nice to evaluate the ObjectiveFunction on the torch object only. One could for instance give the other arguments to the constructor.

Reply:
Just noticed that you did this already for the AcquisitionModel. Why not here, too?